src/bindings/js.ts

```diff
@@ -1367,3 +1367,56 @@ function indentText(text: string, indentLevel: i32, sb: string[], butFirst: bool
     sb.push(text.substring(lineStart));
   }
 }
+
+export function liftRequiresExportRuntime(type: Type): bool {
+  if (!type.isInternalReference) return false;
+  let clazz = type.classReference;
+  if (!clazz) {
+    // functions lift as internref using __pin
+    assert(type.signatureReference);
+    return true;
+  }
+  let program = clazz.program;
+  // flat collections lift via memory copy
+  if (
+    clazz.extends(program.arrayBufferInstance.prototype) ||
+    clazz.extends(program.stringInstance.prototype) ||
+    clazz.extends(program.arrayBufferViewInstance.prototype)
+  ) {
+    return false;
+  }
+  // nested collections lift depending on element type
+  if (
+    clazz.extends(program.arrayPrototype) ||
+    clazz.extends(program.staticArrayPrototype)
```

Comment:
Could you also add Map and Set which will be support in future?

Reply:
Right now these fall into the internref category I think, so should work according to what is supported. Once these are supported via memory copy, it should be trivial to add them here and with it change the behavior. Seems right now it would be asymmetric, covering something that is not actually supported.